pkg/snet/network.go

```diff
@@ -163,32 +204,81 @@ func New(conf Config, eb *appevent.Broadcaster) (*Network, error) {
 		clients.StcpC.SetLogger(logging.MustGetLogger("snet.stcpC"))
 	}
 
-	if conf.NetworkConfigs.STCPR != nil {
-		ar, err := arclient.NewHTTP(conf.NetworkConfigs.STCPR.AddressResolver, conf.PubKey, conf.SecKey)
-		if err != nil {
-			log.WithError(err).Warnf("Failed to connect to address-resolver, STCPR will be disabled")
-		} else {
-			addressResolver = ar
+	if conf.NetworkConfigs.STCPR != nil || conf.NetworkConfigs.STCPH != nil {
+		var (
+			addressResolver   arclient.APIClient
```

Comment:
@evanlinjin @Darkren Would you mind if I work on the reconnection logic of `arclient.httpClient`? Its code is familiar to me and I have some ideas for reconnection logic. Moreover, I'm going to add a UDP connection in https://github.com/SkycoinProject/skywire-mainnet/pull/400. In my opinion, it would make more sense to implement reconnection logic after UDP implementation is finished.

Reply:
@nkryuchkov I'll leave a `TODO` comment then